File: php/features.php
<?php
declare(strict_types=1);

// AdviceSlip SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AdviceSlipFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AdviceSlipBaseFeature();
            case "test":
                return new AdviceSlipTestFeature();
            default:
                return new AdviceSlipBaseFeature();
        }
    }
}

File: py/features.py
# AdviceSlip SDK feature factory

from feature.base_feature import AdviceSlipBaseFeature
from feature.test_feature import AdviceSlipTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AdviceSlipBaseFeature(),
        "test": lambda: AdviceSlipTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
